packages/xod-cli/src/messages.js

```diff
@@ -1,9 +1,174 @@
+
+export const publishCommandDescription = 'publish a library';
+
+export const publishCommandExamples = [
+  'Publish the current project with the version defined in `project.xod`\n' +
+    '$ xodc publish',
+  'Publish a project saved as xodball\n$ xodc publish foo.xodball',
+];
+
+export const publishCommandGetLib = 'Get library';
+
+export const publishCommandGetUser = 'Get user';
+
+export const publishCommandLibGetError = (libname, msg) =>
+  `Can't get library ${libname} because of ${msg}`;
+
+export const publishCommandLibLoading = 'Load library';
+
+export const publishCommandLibPost = 'Publish library';
+
+export const publishCommandLibPostError = msg =>
+  `Can't publish library because of ${msg}`;
+
+export const publishCommandLibPutError = msg =>
+  `Can't create library because of ${msg}`;
+
+export const publishCommandUsage = 'publish [options] [entrypoint]';
+
+export const publishCommandUser404 = username =>
+  `User ${username} doesn't exist`;
+
+export const publishCommandUserError = msg =>
+  `Can't get user because of ${msg}`;
+
+export const publishCommandVersionExists = libpath =>
+  `Version ${libpath} already exists`;
+
+export const quietFlagDescription = 'do not log messages other than errors';
+
+export const resaveCommandDescription =
+  'opens a project and saves it in another location or format';
+
+export const resaveCommandExamples = [
+  `Exports the current multifile project to a xodball\n` +
+    `$ xodc resave . -o ~/foo.xodball\n`,
+  `Outputs the current multifile project as a xodball to stdout\n` +
+    `$ xodc resave\n`,
+  `Resaves one xodball into another (useful for applying migrations)\n` +
+    `$ xodc resave foo.xodball -o bar.xodball\n`,
+  `Converts a xodball to a multifile project\n` +
+    `$ xodc resave foo.xodball -o /some/new/dir\n`,
+];
+
+export const resaveCommandProjectLoading = 'Project loading...';
+
+export const resaveCommandSaving = 'Saving...';
+
+export const resaveCommandUsage = 'resave [options] [entrypoint]';
+
+export const resaveCommandOutputFlagDescription =
+  'xodball or multifile directory output path, defaults to stdout';
+
+export const resaveSuccess = p => `Project resaved into ${p}`;
+
+export const tabtestCommandDescription = 'tabtest project';
+
+export const tabtestCommandUsage = 'tabtest [options] [project]';
+
+export const tabtestCommandGeneratingCpp = 'Generating C++ code...';
+
+export const tabtestCommandExamples = [
+  `Build tabtests for project in current working directory\n` +
+    `$ xodc tabtest`,
+  `Specify target directory and project, only generate tests\n` +
+    `$ xodc tabtest --no-build --output-dir=/tmp/xod-tabtest ./workspace/__lib__/xod/net`,
+];
+
+export const tabtestCommandExitedWithCode = (cmd, code) =>
+  `${cmd} exited with code ${code}`;
+
+export const tabtestCommandNobuildFlagDescription = 'do not build';
+
+export const tabletestCommandLoadingProject = 'Loading project...';
```

Comment:
Some messages are variables, others are functions.
Maybe better to do all messages to be functions?

Reply:
May be it's better to use some common i18n library?

Reply:
No. Use the established convention. All formatters should be functions returning an object with the following fields:

- title
- note
- solution

`note` and `solution` are optional and written in normal English. The `title` is a very short headlinesse.

Once you got a message object, output it to the console, possibly with bells and whistles like colors, funky unicode, etc.

The important point is using the defined structure. It helps a user to quickly find a solution for an error.